bodo/io/iceberg/__init__.py

```diff
@@ -53,11 +53,10 @@
 
 def get_iceberg_pq_dataset(
     conn: str,
-    database_schema: str,
-    table_name: str,
+    table_id: str,
     typing_pa_table_schema: pa.Schema,
     str_as_dict_cols: list[str],
-    iceberg_filter: str | None = None,
+    iceberg_filter: BooleanExpression,
```

Comment:
BooleanExpression is set of filter classes in PyIceberg. Nicer than using our own custom repr.